sdk/resource/auto.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+var (
+	// ErrMissingResource is returned by a detector when source information
+	// is unavailable for a Resource.
+	ErrMissingResource error
+	// ErrPartialResource is returned by a detector when complete source
+	// information for a Resource is unavailable or the source information
+	// contains invalid values that are omitted from the returned Resource.
+	ErrPartialResource error
```

Comment:
Sorry, I failed to point out my last suggestion was a starting point.

We probably want to add some error message here.

```suggestion
	ErrMissingResource = errors.New("missing resource")
	// ErrPartialResource is returned by a detector when complete source
	// information for a Resource is unavailable or the source information
	// contains invalid values that are omitted from the returned Resource.
	ErrPartialResource = errors.New("partial resource")
```

Reply:
I see. By the way, I guess ErrMissingResource can be used when the resource returned by the detector is empty?

Reply:
> I guess ErrMissingResource can be used when the resource returned by the detector is empty?

That was the idea, though looking at the implementation it might make sense to not include it if we are not going to test or handle this type of error.